tests/test_trace_exporter.py

```diff
@@ -0,0 +1,123 @@
+import unittest
+
+from unittest.mock import patch
+
+
+from opentelemetry import trace as trace_api
+from opentelemetry.sdk import trace
+from opentelemetry.ext.lightstep import LightStepSpanExporter
+from opentelemetry.sdk.trace.export import SpanExportResult
+
+
+class MockTracer:
+    """ Used to mock lightstep.Tracer """
+
+    def __init__(
+        self,
+        component_name,
+        access_token,
+        collector_host,
+        collector_port,
+        collector_encryption,
+        verbosity,
+        use_http=False,
+        use_thrift=False,
+    ):
+        self.name = component_name
+        self.token = access_token
+        self.host = collector_host
+        self.port = collector_port
+        self.encryption = collector_encryption
+        self.verbosity = verbosity
+        self.use_http = use_http
+        self.use_thrift = use_thrift
+        self.spans = []
+
+    def record(self, span):
+        self.spans.append(span)
+
+    def flush(self):
+        pass
+
+
+class TestLightStepSpanExporter(unittest.TestCase):
+    @patch("lightstep.Tracer", MockTracer)
+    def test_constructor_default(self):
+        """Test the default values assigned by constructor."""
+        name = "my-service-name"
+        host = "ingest.lightstep.com"
+        port = 443
+        encryption = "tls"
+        verbosity = 0
+        token = ""
+        exporter = LightStepSpanExporter(name)
+
+        self.assertEqual(exporter.tracer.name, name)
+        self.assertEqual(exporter.tracer.host, host)
+        self.assertEqual(exporter.tracer.port, port)
+        self.assertEqual(exporter.tracer.encryption, encryption)
+        self.assertEqual(exporter.tracer.verbosity, verbosity)
+        self.assertEqual(exporter.tracer.token, token)
+
+    @patch("lightstep.Tracer", MockTracer)
+    def test_export(self):
+        # pylint: disable=invalid-name
+        self.maxDiff = None
+
+        span_names = ("test1", "test2", "test3")
+        trace_id = 0x6E0C63257DE34C926F9EFCD03927272E
+        trace_id_high = 0x6E0C63257DE34C92
+        trace_id_low = 0x6F9EFCD03927272E
+        span_id = 0x34BF92DEEFC58C92
+        parent_id = 0x1111111111111111
+        other_id = 0x2222222222222222
+
+        base_time = 683647322 * 10 ** 9  # in ns
+        start_times = (
+            base_time,
+            base_time + 150 * 10 ** 6,
+            base_time + 300 * 10 ** 6,
+        )
+        durations = (50 * 10 ** 6, 100 * 10 ** 6, 200 * 10 ** 6)
+        end_times = (
```

Comment:
`end_times` is unused. That means that `start_times` and `durations` can be removed as well. Once they are removed, `base_time` can be removed and its value used directly in the only remaining place where that variable is used.

Reply:
added tests for start times and duration,